src/atom_collection.rs

```diff
@@ -438,6 +670,15 @@ fn last_leaf(node: Node) -> Node {
     }
 }
 
+static mut COUNTER: usize = 0;
```

Comment:
Could we not put this in `AtomCollection` instead, and get rid of `static` and `unsafe`?

Reply:
I didn't think about that ; it's much better your way.